src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -0,0 +1,115 @@
+import org.terasology.logic.characters.CharacterSoundComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.health.event.DoDamageEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.physics.events.MovedEvent;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class HypothermiaSystem extends BaseComponentSystem {
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Context context;
+
+    private float thresholdHeight = 60f;
+    private int healthDecreaseInterval = 20000;
+    private int initialDelay = 5000;
+    private int healthDecreaseAmount = 15;
+    private Random random = new FastRandom();
+
+
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void observeDangerZone(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
+        //TODO: react on OnBiomeChangedEvent to handle the danger zone
+        float height = event.getPosition().getY();
+        float lastHeight = height - event.getDelta().getY();
+        if (height > thresholdHeight && lastHeight <= thresholdHeight) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+        if (height < thresholdHeight && lastHeight >= thresholdHeight) {
+            if(player.hasComponent(HypothermiaComponent.class)){
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onHypothermia(OnAddedComponent event, EntityRef player, HypothermiaComponent hypothermia){
+        if(player.hasComponent(HypothermiaComponent.class)){
+            delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, initialDelay, healthDecreaseInterval);
+        }
+    }
+    @ReceiveEvent(components = {LocationComponent.class})
+    public void onPeriodicFrostbite(PeriodicActionTriggeredEvent event, EntityRef player) {
+        if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
+            // Check to see if health should be decreased
+            LocationComponent location = player.getComponent(LocationComponent.class);
+            float height = location.getLocalPosition().getY();
+            if (height > thresholdHeight) {
+                applyFrostbiteDamagePlayer(player);
+                // Stun the player for 500 ms
+                applyStunEffect(player, 1000);
+                playFrostbiteSound(player);
+            }
```

Comment:
As the periodic frostbite is only enabled when the `HypothermiaComponent` is added and that one already check the "danger zone", you don't need to check it again here.
```suggestion
            applyFrostbiteDamagePlayer(player);
            applyStunEffect(player, 1000);
            playFrostbiteSound(player);
```
Also, the comment concerning the stunning was mentioning the wrong duration and is generally not necessary as the method name describes well enough what's happening :wink: